src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3239,14 +3245,42 @@ internal virtual ImmutableArray<LabelSymbol> Labels
                 // (i.e. doesn't work with "params" or optional parameters).
                 if (!method.Parameters.Any())
                 {
-                    candidateMethods.Add((MethodSymbol)member);
+                    candidateMethods.Add(method);
                 }
             }
 
-            MethodSymbol patternMethod = PerformPatternOverloadResolution(patternType, candidateMethods, syntaxExpr, warningsOnly, diagnostics, syntaxTree, messageID);
-
+            patternMethod = PerformPatternOverloadResolution(patternType, candidateMethods, syntaxExpr, warningsOnly, diagnostics, syntaxTree, messageID);
             candidateMethods.Free();
 
+
+            // if we failed to find a method via normal lookup, see if there is an extension method that satisifies it
+            if (patternMethod == null && searchExtensionMethods)
+            {
+                var extLookupResult = LookupResult.GetInstance();
+                this.LookupExtensionMethods(extLookupResult, methodName, 0, lookupOptions, ref useSiteDiagnostics);
+
+                // if there are multiple extension methods found it is ambiguous which one to use 
+                if (extLookupResult.IsMultiViable && extLookupResult.Symbols.Count > 1)
+                {
+                    diagnostics.Add(ErrorCode.WRN_PatternIsAmbiguous, syntaxExpr.Location, patternType, messageID.Localize(),
```

Comment:
Seems like the instance lookup just tries to find something with no parameters. Why can't we do that here?

This will probably break for people using both a `Dispose()` and a `Dispose(bool)` method, which is a pretty common pattern.